### PR TITLE
Allow custom root directory for config

### LINKED ISSUE #5779: Make astropy.config directory configurable by packages that use astropy_helpers
Packages that depend on `astropy` and `astropy_helpers` may not necessarily want to be affiliated packages, so they may not want to put their configuration in `~/.astropy/package_name`. We could make the config directory default to that, but allow it to be configurable to, e.g., `~/.package_name`.

### PR DESCRIPTION
**Pull request by Cadair**
This is an adoption of #5828 because I think I have more reason to want it than @adrn.

I have rebased this for 3.0+ (so it's Python 3 only now) and I have also opened astropy/package-template#290 which demonstrates the changes needed to use this in the template. Also I have used this successfully in DKISTDC/dkist#3 to prove it works.

@eteq opinions?

EDIT: Fix #5779 , close #5828

### CODE REVIEW

File: astropy/config/configuration.py [hunk 1]
```diff
@@ -92,6 +92,12 @@ class Conf(_config.ConfigNamespace):
                 aliases=['astropy.utils.console.USE_COLOR'])
         conf = Conf()
     """
+
+    rootname = 'astropy'
```
Comment (pllim) [inline astropy/config/configuration.py]: This is defined again in `ConfigItem`. Is this necessary here? Seems weird to define this twice separately.

File: astropy/utils/data.py [hunk 2]
```diff
@@ -967,6 +967,8 @@ def download_file(remote_url, cache=False, show_progress=True, timeout=None):
         The timeout, in seconds.  Otherwise, use
         `astropy.utils.data.Conf.remote_timeout`.
 
+    pkgname : TODO
```
Comment (pllim) [inline astropy/utils/data.py]: This docstring is incomplete.

File: astropy/utils/data.py [hunk 3]
```diff
@@ -1112,10 +1114,10 @@ def is_url_in_cache(url_key):
 
 
 def _do_download_files_in_parallel(args):
-    return download_file(*args)
+    return download_file(*args, show_progress=False)
 
 
-def download_files_in_parallel(urls, cache=True, show_progress=True,
+def download_files_in_parallel(urls, cache=False, show_progress=True,
```
Comment (pllim) [inline astropy/utils/data.py]: This change in behavior seems unrelated to this PR.

File: astropy/utils/data.py [hunk 4]
```diff
@@ -1112,10 +1114,10 @@ def is_url_in_cache(url_key):
 
 
 def _do_download_files_in_parallel(args):
-    return download_file(*args)
+    return download_file(*args, show_progress=False)
```
Comment (pllim) [inline astropy/utils/data.py]: This change in behavior seems unrelated to this PR.

File: astropy/utils/data.py [hunk 5]
```diff
@@ -1168,6 +1170,10 @@ def download_files_in_parallel(urls, cache=True, show_progress=True,
     else:
         progress = io.BytesIO()
 
+    if timeout is None:
```
Comment (pllim) [inline astropy/utils/data.py]: This change in behavior seems unrelated to this PR.

**Comment by Cadair** [comment]
looks like @pllim just found a whole bunch of my rebase fails 😆

File: astropy/config/configuration.py (hunk 1 above)
Comment (Cadair) [reply astropy/config/configuration.py]: Different bits of the machinery use `ConfigItem` and `ConfigNamespace` I could not see a simple way of sharing the value.

Comment (Cadair) [reply astropy/config/configuration.py]: although maybe this isn't actually used anywhere... 😕

**Comment by Cadair** [comment]
I think all the major wrinkles in this are iorned out. Review please!

**Comment by Cadair** [comment]
That's curious, I can't reproduce the Circle CI fail locally. That is a post-lunch problem.

**Comment by adrn** [comment]
Given that I wrote some fraction of this code, it would be good for @eteq to review this 😄

File: astropy/config/paths.py [hunk 6]
```diff
@@ -71,21 +71,31 @@ def _find_home():
...
         else:
             raise OSError('Could not find a home directory to search for '
-                          'astropy config dir - are you on an unspported '
+                          'astropy config dir - are you on an unsupported '
                           'platform?')
     return homedir
 
 
-def get_config_dir(create=True):
+def get_config_dir(rootname, create=True):
```
Comment (eteq) [inline astropy/config/paths.py]: This is a public API function so this is a breaking API change.  I think `rootname` needs to default to `astropy`, and be the second argument.  I realize this isn't used that much, but I think it *is* used by some affiliated packages that use tricks to do something else.

File: astropy/config/tests/test_configs.py [hunk 7]
```diff
@@ -17,56 +17,59 @@
 
 
 def test_paths():
-    assert 'astropy' in paths.get_config_dir()
-    assert 'astropy' in paths.get_cache_dir()
+    assert 'astropy' in paths.get_config_dir('astropy')
+    assert 'astropy' in paths.get_cache_dir('astropy')
```
Comment (eteq) [inline astropy/config/tests/test_configs.py]: Related to the above: this test has to be left unchanged to not be a public API-breaking change.  Same with all the stuff below in this file.

File: astropy/config/paths.py [hunk 8]
```diff
@@ -108,16 +118,16 @@ def get_config_dir(create=True):
...
             if os.path.exists(xchpth):
                 return os.path.abspath(xchpth)
             else:
                 linkto = xchpth
-    return os.path.abspath(_find_or_create_astropy_dir('config', linkto))
+    return os.path.abspath(_find_or_create_root_dir('config', linkto, rootname))
 
 
-def get_cache_dir():
+def get_cache_dir(rootname):
```
Comment (eteq) [inline astropy/config/paths.py]: Same deal here with it being an API-breaking change.  Needs to default to `'astropy'`

**Comment by astrofrog** [comment]
There was a bug in the bot, it should not have been closed without the warning so re-opening.
